### PR TITLE
Updated target frameworks to netcoreapp1.1,netstandard1.6, and net4.6…

### PR DESCRIPTION
**Pull request by agatlin**
…. Went through and tweaked every code file for performance. (Generally, this consisted of inlining variables, getting rid of string concatenation, compiling regexes, etc.)Replaced HtmlAgilityPack with HtmlAgilityPack.Core 1.5.x.

### CODE REVIEW

**Comment by mysticmind** [comment]
@agatlin CI build is failing for your PR. You will see a little red cross in the PR, if you hover, it will show links to Appveyor CI and Travis CI, both the builds are failing 

Few observations/comments, 
1. Please don't add .netcoreapp1.1 target to ReverseMarkdown project. As I had clearly told you that .Net Standard libraries are compatible to work with .Net Core 1.0 and 1.1. Hence an explicit .netcoreapp1.1 target is not required. Also you have removed net45 target.

2. In the ReverseMarkdown.Test project, please leave the version target version to netcoreapp1.0, net45 and net46. This will automatically build and test for netstandard1.6.  And netcoreapp1.1 is not required since netcore1.0 is the least common denominator from a test project perspective.

Note that I have clearly explained the above points in the issues which you had participated and also the choice of using my other library MysticMind.HtmlAgilityPack library which you have changed as well to your preferred one. Please revert it to use MysticMind.HtmlAgilityPack.

I don't seem comfortable with this PR due to the above observations. And it is also going to be very difficult to review your changes since you have changed every file possible.

Request you to first take a look at my comments above, fix those and get the CI builds green. Please also provide some data points around how much the performance have improved with your changes to really weigh in if the PR is worth pursuing.

**Comment by agatlin** [comment]
I am aware of your comments, but some of your points are incorrect. 

.Net Standard 1.6 is not necessarily fully compatible with .Net Core 1.1--only with .Net Core 1.0. There are quite a number of features in .Net Core 1.1 which are are not available in .Net Standard 1.6. If you are targeting .Net Standard 1.6, you should be targeting .NET 4.61 framework. If you want to target .NET 4.5, you should be targeting .Net Standard 1.0. 

For more information, look here . https://docs.microsoft.com/en-us/dotnet/standard/library.

Reflection is flaky as Hell in .Net Core 1.0. It is a bit better in .Net Core 1.1. It will be much better in the next release.

Speaking of reflection, I actually thought about refactoring the code to remove reflection altogether and access the conversion routines as static or extension methods. Performance would actually be a little better and there would be less compatibility issues. 

I changed your dependencies because they were compatibility issues with them and because the version of HTML Agility Pack I provided is already production quality, already released for .NET Core, and already used by many people. With all due respect, your hacked internal version is not nearly as thoroughly tested--and there were compatibility issues when I attempted to use it with .NET Core 1.1.

Performance improvements are more than 300% for small datasets and even higher for large data sets. Unit tests that were taking 135-145 ms running under .NET 4.5 are now completing in about 35ms under .NET Core 1.1. For tests that use compiled regular expressions, speedups are even more significant after first use (once the Regex has compiled.)

I know the code builds as configured and all unit Tests pass. I don't know about your Appveyor CI and Travis CI.  Apologies, but I am generally uninterested in trying to figure out your build configuration. I have a lot of work to do, and I just don't have time to focus on that. 

You can use my code or not. I modified the code because I actually needed it to be production ready.  (By the way, there is still virtually no error handling in the code--which is something I will also be adding.)

What I gave you is faster and better than what was originally written. 

If it is not what you want, then ignore it. I can always further refactor the code and put it out as a new package. I have already significantly changed it from the original version anyway. With the architectural enhancements I would like to make and the features I want to add, it will essentially be a different product. I am sure most users would prefer to access a better performing package with additional functionality and error handling anyway.

I was just trying to help improve your code. I apologize for trying to help you make your package better.

**Comment by mysticmind** [comment]
> I am aware of your comments, but some of your points are incorrect.

I beg to disagree with some of your view points/comments since they are certainly incorrect. 


> .Net Standard 1.6 is not necessarily fully compatible with .Net Core 1.1--only with .Net Core 1.0. There are quite a number of features in .Net Core 1.1 which are are not available in .Net Standard 1.6. If you are targeting .Net Standard 1.6, you should be targeting .NET 4.61 framework. If you want to target .NET 4.5, you should be targeting .Net Standard 1.0.

ReverseMarkdown library targeting .Net Standard 1.6 and the API surface used by the library does work fine with .Net Core 1.0 and .Net core 1.1, even though .Net Standard 1.6 may not necessarily be fully compatible with .Net Core 1.1. I don't see a reason to specifically target the library for .Net Core 1.1

Note that the project has multi-targeting hence specific assembly artifacts are created for .Net 4.5, .Net 4.6 and .Net Standard 1.6 respectively.

> Reflection is flaky as Hell in .Net Core 1.0. It is a bit better in .Net Core 1.1. It will be much better in the next release.

The reflection API available in .Net Standard 1.6 is more than sufficient to handle the requirements for ReverseMarkdown.

> I changed your dependencies because they were compatibility issues with them and because the version of HTML Agility Pack I provided is already production quality, already released for .NET Core, and already used by many people. With all due respect, your hacked internal version is not nearly as thoroughly tested--and there were compatibility issues when I attempted to use it with .NET Core 1.1.

MysticMind.HtmlAgilityPack is not a hacked internal version. It is production quality with automated CI unit tests and is up in Nuget for close to an year now and also used by many. It does work fine with .Net Core 1.1 and have tested it.

> I know the code builds as configured and all unit Tests pass. I don't know about your Appveyor CI and Travis CI. Apologies, but I am generally uninterested in trying to figure out your build configuration. I have a lot of work to do, and I just don't have time to focus on that.

When you contribute to an OSS project, it is imperative to ensure the build and all unit tests pass as expected. Checking the CI build failure log is a very important step for contributors to ensure functionality works as expected. It does not necessarily require you to understand the underpinning of the CI build configuration.

> You can use my code or not. I modified the code because I actually needed it to be production ready. (By the way, there is still virtually no error handling in the code--which is something I will also be adding.)

Note that the project is available for close to 2 years with published Nuget packages and it is certainly production ready since we use it within our products in production. Your mileage may vary based on the parameters you choose to qualify something as production ready.

In summary, we have divergent views. I do appreciate your time and interest on this project. But have taken a call to close this without merging this PR.
